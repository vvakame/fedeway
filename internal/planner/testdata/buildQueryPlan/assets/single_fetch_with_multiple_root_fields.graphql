# schema: supergraphSdl.graphqls

# should use a single fetch when requesting multiple root fields from the same service
query {
    topProducts {
        name
    }
    product(upc: "1") {
        name
    }
}
