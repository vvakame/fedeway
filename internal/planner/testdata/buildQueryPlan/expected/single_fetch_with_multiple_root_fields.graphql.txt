QueryPlan {
	Sequence {
		Fetch(service: "product") {
			query {
				topProducts {
					__typename
					... on Book {
						__typename
						isbn
					}
					... on Furniture {
						name
					}
				}
				product(upc: "1") {
					__typename
					... on Book {
						__typename
						isbn
					}
					... on Furniture {
						name
					}
				}
			}
		},
		Parallel {
			Sequence {
				Flatten(path: "topProducts.@") {
					Fetch(service: "books") {
						{
							... on Book {
								__typename
								isbn
							}
						} =>
						{
							... on Book {
								__typename
								isbn
								title
								year
							}
						}
					},
				},
				Flatten(path: "topProducts.@") {
					Fetch(service: "product") {
						{
							... on Book {
								__typename
								isbn
								title
								year
							}
						} =>
						{
							... on Book {
								name
							}
						}
					},
				},
			},
			Sequence {
				Flatten(path: "product") {
					Fetch(service: "books") {
						{
							... on Book {
								__typename
								isbn
							}
						} =>
						{
							... on Book {
								__typename
								isbn
								title
								year
							}
						}
					},
				},
				Flatten(path: "product") {
					Fetch(service: "product") {
						{
							... on Book {
								__typename
								isbn
								title
								year
							}
						} =>
						{
							... on Book {
								name
							}
						}
					},
				},
			},
		},
	},
}